tests/test_core/test_new_moltype.py

```diff
@@ -167,28 +167,115 @@ def test_gap_index_constant(moltype):
 def test_get_degenerate_positions(data_type, moltype):
     seq = make_typed("ASA", data_type, moltype)
     got = moltype.get_degenerate_positions(seq)
-    expect = numpy.array([False, True, False], dtype=bool)
-    assert numpy.array_equal(got, expect)
+    expect = [1]
+    assert got == expect
 
     seq = make_typed("A-SA", data_type, moltype)
     got = moltype.get_degenerate_positions(seq)
-    expect = numpy.array([False, True, True, False], dtype=bool)
-    assert numpy.array_equal(got, expect)
+    expect = [1, 2]
+    assert got == expect
 
     got = moltype.get_degenerate_positions(seq, include_gap=False)
-    expect = numpy.array([False, False, True, False], dtype=bool)
-    assert numpy.array_equal(got, expect)
+    expect = [2]
+    assert got == expect
 
     seq = make_typed("BAB", data_type, moltype)
     got = moltype.get_degenerate_positions(seq)
-    expect = numpy.array([True, False, True], dtype=bool)
-    assert numpy.array_equal(got, expect)
+    expect = [0, 2]
+    assert got == expect
 
     seq = make_typed("---", data_type, moltype)
     got = moltype.get_degenerate_positions(seq)
-    expect = numpy.array([True, True, True], dtype=bool)
+    expect = [0, 1, 2]
+    assert got == expect
 
     seq = make_typed("", data_type, moltype)
     got = moltype.get_degenerate_positions(seq)
-    expect = numpy.array([], dtype=bool)
-    assert numpy.array_equal(got, expect)
+    expect = []
+    assert got == expect
+
+
+def test_resolve_ambiguity_nucs():
+    got = new_moltype.DNA.resolve_ambiguity("AT?", allow_gap=False)
+    assert len(got) == 4
+    assert len(got[0]) == 3
+
+
+def test_resolve_ambiguity_codons():
```

Comment:
need to implement a codon alphabet class, please remind me when I get back. Add a comment here to use new_genetic_code when that job is done.